Marlin/src/pins/pins_BIQU_BQ111-A4.h

```diff
@@ -0,0 +1,154 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+/**
+ * BIQU BQ111-A4
+ *
+ * Applies to the following boards:
+ *
+ *  BOARD_BIQU_BQ111_A4 (Hotend, Fan, Bed)
+ *
+ */
+
+#ifndef TARGET_LPC1768
+  #error "Oops!  Make sure you have Re-Arm selected."
+#endif
+
+#ifndef BOARD_NAME
+  #define BOARD_NAME "BIQU BQ111-A4"
+#endif
+
+#define LARGE_FLASH true
+
+/**
+ * Unused
+ */
+//#define PIN_P2_10         P2_10  // IBOOT-1
+//#define PIN_P0_27         P0_27  // Onboard SD Detect
+
+
+/**
+ * Limit Switches
+ */
+#define X_MIN_PIN         P1_24  //10k pullup to 3.3V, 1K series
+#define X_MAX_PIN         P1_25  //10k pullup to 3.3V, 1K series
+#define Y_MIN_PIN         P1_26  //10k pullup to 3.3V, 1K series
+#define Y_MAX_PIN         P1_27  //10k pullup to 3.3V, 1K series
+#define Z_MIN_PIN         P1_28  //10k pullup to 3.3V, 1K series
+#define Z_MAX_PIN         P1_29  //10k pullup to 3.3V, 1K series
+
+
+/**
+ * Steppers
+ */
+#define X_STEP_PIN         P2_0
+#define X_DIR_PIN          P0_5
+#define X_ENABLE_PIN       P0_4
+
+#define Y_STEP_PIN         P2_1
+#define Y_DIR_PIN          P0_11
+#define Y_ENABLE_PIN       P0_10
+
+#define Z_STEP_PIN         P2_2
+#define Z_DIR_PIN          P0_20
+#define Z_ENABLE_PIN       P0_19
+
+#define E0_STEP_PIN        P2_3
+#define E0_DIR_PIN         P0_22
+#define E0_ENABLE_PIN      P0_21
+
+
+/**
+ * Temperature Sensors
+ *  3.3V max when defined as an analog input
+ */
+#define TEMP_0_PIN          0  // A0 (T0)
+#define TEMP_BED_PIN        1  // A1 (T1)
+
+
+/**
+ * Heaters / Fans
+ */
+#define HEATER_0_PIN       P2_7
+#define HEATER_BED_PIN     P2_5
+#define FAN_PIN            P2_4
+
+
+/**
+ * LCD / Controller
+ *
+ * REPRAP_DISCOUNT_SMART_CONTROLLER is not supported due to the lack of LCD_PINS_D5,
+ * LCD_PINS_D6 or LCD_PINS_D7 in the EXP1 connector.
+ * 
+ * A remote SD card is not supported as the pins routed to EXP2 are the same as used
+ * for the onboard SD card, and a chip select signal is not provided for the remote
+ * SD card.
+ */
+#if ENABLED(ULTRA_LCD)
+
+  #define BEEPER_PIN          P1_31  // EXP1-1
+
+  #define BTN_EN1             P3_26  // EXP2-3
+  #define BTN_EN2             P3_25  // EXP2-5
+  #define BTN_ENC             P1_30  // EXP1-2
+
+  #define SD_DETECT_PIN       P0_27  // EXP2-7
+  #define LCD_PINS_RS         P0_16  // EXP1-4
+  #define LCD_PINS_ENABLE     P0_18  // (MOSI) EXP1-3
+  #define LCD_PINS_D4         P0_15  // (SCK)  EXP1-5
+
+  #if ENABLED(REPRAP_DISCOUNT_SMART_CONTROLLER) && !ENABLED(DOGLCD)
+    #error "REPRAP_DISCOUNT_SMART_CONTROLLER is not supported by the BIQU BQ111-A4"
+  #endif
+
+  #if ENABLED(SDSUPPORT)
+    #error "SDSUPPORT is not supported by the BIQU BQ111-A4"
+  #endif
```

Comment:
It `SDSUPPORT` supported outside of `ENABLED(ULTRA_LCD)`? If not, this should be moved outside of `ENABLED(ULTRA_LCD)`.

Reply:
That's a good point. The only reason I put it there is because it seems like the pins for accessing the SD cards are only defined in the `ENABLED(ULTRA_LCD)` sections of other pins files. However, I don't see any sort of sanity check to ensure that `ULTRA_LCD` is enabled if `SDSUPPORT` is also enabled.

Reply:
It is hoped that users can install an SD drive without an LCD, if they wish. I haven't seen this in action.

Reply:
I went ahead and clarified the error that the `SDSUPPORT` can't be used when an LCD is in use. I also added a section for when just `SDSUPPORT` is enabled to define the SPI pins to available pins on the EXP-1 connector. The user would have to create a custom cable to interface the SD drive with the controller board, but I think that's the best I can do given the limited capability of this board.